integration-tests/deployment/keystone/types.go

```diff
@@ -0,0 +1,306 @@
+package keystone
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	chainsel "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink/integration-tests/deployment"
+	"github.com/smartcontractkit/chainlink/integration-tests/deployment/clo/models"
+	v1 "github.com/smartcontractkit/chainlink/integration-tests/deployment/jd/node/v1"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/capabilities_registry"
+	kcr "github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/capabilities_registry"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/p2pkey"
+)
+
+var (
+	CapabilitiesRegistry deployment.ContractType = "CapabilitiesRegistry"
+	KeystoneForwarder    deployment.ContractType = "KeystoneForwarder"
+	OCR3Capability       deployment.ContractType = "OCR3Capability"
+)
+
+type deployResponse struct {
+	Address common.Address
+	Tx      common.Hash // todo: chain agnostic
+	Tv      deployment.TypeAndVersion
+}
+
+type deployRequest struct {
+	Chain deployment.Chain
+}
+
+type DonNode struct {
+	Don  string
+	Node string // not unique across environments
+}
+
+type CapabilityHost struct {
+	NodeID       string // globally unique
+	Capabilities []capabilities_registry.CapabilitiesRegistryCapability
+}
+
+type Nop struct {
+	capabilities_registry.CapabilitiesRegistryNodeOperator
+	NodeIDs []string // nodes run by this operator
+}
+
+// ocr2Node is a subset of the node configuration that is needed to register a node
```

Comment:
you sure you can't go straight from OffchainClient -> Node https://github.com/smartcontractkit/chainlink/blob/5c51b27bb1350645d08afab2e98e56ba84ae5c28/integration-tests/deployment/environment.go#L173 -> Capability registration? We did that here https://github.com/smartcontractkit/chainlink/blob/9efabb039f78c1890a4a3749fb4986d6b7947733/integration-tests/deployment/ccip/deploy_home_chain.go#L200

Reply:
added a ticket and todo that can be address when we fully migrate to JD.

this struct is an internal helper, so there is no risk to refactoring it later.